lib/vend/oauth2/auth_code.rb

```diff
@@ -0,0 +1,48 @@
+require 'oauth2'
+
+module Vend
+  module Oauth2
+
+    class AuthCode
+
+      DEFAULT_OPTIONS = {}
+      AUTHORIZE_URL = '/connect'
+      TOKEN_URL = '/api/1.0/token'
+
+      attr_accessor :store, :client_id, :secret, :redirect_uri
+
+      def initialize(store, client_id, secret, redirect_uri, options = {})
+        @store = store
+        @client_id = client_id
+        @secret = secret
+        @redirect_uri = redirect_uri
+        @options = DEFAULT_OPTIONS.merge(options)
+      end
+
+      def authorize_url
+        get_oauth2_client.auth_code.authorize_url(:redirect_uri => redirect_uri)
+      end
+
+      def get_token(code)
```

Comment:
Probably a good idea to make this method name indicative that you're using the code callback OAuth grant flow. Perhaps `get_token_from_code(code)` OR `token_from_code(code)`?